### PR TITLE
add TLS support for OLS app server

### PR DESCRIPTION
**Pull request by vbelouso**
## Description

add TLS support for OLS app server

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

https://issues.redhat.com/browse/OLS-242

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: api/v1alpha1/olsconfig_types.go [hunk 1]
```diff
@@ -56,6 +56,9 @@ type OLSSpec struct {
 	// +kubebuilder:validation:Enum=DEBUG;INFO;WARNING;ERROR;CRITICAL
 	// +kubebuilder:default=INFO
 	LogLevel string `json:"logLevel,omitempty"`
+	// Disable TLS for OLS server. Default: "false"
+	// +kubebuilder:default=false
+	DisableTLS bool `json:"disableTLS,omitempty"`
```
Comment (xrajesh) [inline api/v1alpha1/olsconfig_types.go:61]: @vbelouso I dont think we need this control for the user. @bparees you want this for any reason ?

File: internal/controller/ols_app_server_assets.go [hunk 2]
```diff
@@ -93,10 +93,18 @@ func (r *OLSConfigReconciler) generateOLSConfigMap(cr *olsv1alpha1.OLSConfig) (*
...
 		ConversationCache: conversationCache,
+		TLSConfig: TLSConfig{
+			TLSCertificatePath: path.Join(OLSAppCertsMountRoot, OLSCertsSecretName, "tls.crt"),
+			TLSKeyPath:         path.Join(OLSAppCertsMountRoot, OLSCertsSecretName, "tls.key"),
+		},
 	}
 
 	devConfig := DevConfig{
-		DisableTLS: true,
+		DisableTLS: cr.Spec.OLSConfig.DisableTLS,
```
Comment (raptorsun) [inline internal/controller/ols_app_server_assets.go:103]: let's add an option to disable authentication, too. We have not yet figured out how to pass an authentication header from the console plugin. 
https://github.com/openshift/lightspeed-service/blob/main/examples/olsconfig.yaml#L68

Comment (xrajesh) [reply internal/controller/ols_app_server_assets.go:103]: I am updating this PR  @raptorsun

File: api/v1alpha1/olsconfig_types.go (hunk 1 above)
Comment (bparees) [reply api/v1alpha1/olsconfig_types.go:61]: nope, you can delete it.  OLS, when managed by the operator, should always secure its endpoints with TLS

**Comment by vbelouso** [comment]
Closed in favor #45

**Pull request closed without merging**